cdap-gateway/src/main/java/io/cdap/cdap/gateway/router/NettyRouter.java

```diff
@@ -106,7 +107,11 @@ public NettyRouter(CConfiguration cConf, SConfiguration sConf, @Named(Constants.
     this.hostname = hostname;
     this.serviceLookup = serviceLookup;
     this.securityEnabled = cConf.getBoolean(Constants.Security.ENABLED, false);
-    this.tokenValidator = tokenValidator;
+    if (SecurityUtil.isManagedSecurity(cConf)) {
```

Comment:
This seems awkward that you request for a parameter but actually not using it. What's the harm of just starting it anyway?

Reply:
Whoops I actually just noticed I did that, fixed